nav2_collision_monitor/src/circle.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2022 Samsung Research Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_collision_monitor/circle.hpp"
+
+#include <math.h>
+#include <cmath>
+#include <exception>
+
+#include "nav2_util/node_utils.hpp"
+
+namespace nav2_collision_monitor
+{
+
+Circle::Circle(
+  const nav2_util::LifecycleNode::WeakPtr & node,
+  const std::string & polygon_name,
+  const std::string & base_frame_id,
+  const double simulation_time_step)
+: PolygonBase::PolygonBase(node, polygon_name, base_frame_id, simulation_time_step),
+  radius_(0.0)
+{
+  RCLCPP_INFO(logger_, "[%s]: Creating Circle", polygon_name_.c_str());
+}
+
+Circle::~Circle()
+{
+  RCLCPP_INFO(logger_, "[%s]: Destroying Circle", polygon_name_.c_str());
+}
+
+void Circle::getPolygon(std::vector<Point> & poly)
+{
+  // Number of polygon points. More edges means better approximation.
+  const double polygon_edges = 16;
+  // Increment of angle during points position calculation
+  double angle_increment = 2 * M_PI / polygon_edges;
+
+  // Clear polygon before filling
+  poly.clear();
+
+  // Making new polygon looks like a circle
+  Point p;
+  for (double angle = 0.0; angle < 2 * M_PI; angle += angle_increment) {
+    p.x = radius_ * std::cos(angle);
+    p.y = radius_ * std::sin(angle);
+    poly.push_back(p);
+  }
+  // Last point should match the first
+  p.x = radius_;
+  p.y = 0.0;
+  poly.push_back(p);
+}
+
+int Circle::getPointsInside(const std::vector<Point> & points)
+{
+  int num = 0;
+  for(Point point : points) {
+    if (point.x*point.x + point.y*point.y < radius_squared_) {
+      num++;
+    }
+  }
+
+  return num;
+}
+
+bool Circle::getParameters(std::string & polygon_topic) {
+  auto node = node_.lock();
+  if (!node) {
+    throw std::runtime_error{"Failed to lock node"};
+  }
+
+  PolygonBase::getParameters(polygon_topic);  // Will always return true
+
+  try {
```

Comment:
Again, no try/catch needed here. Remove and remove return type bool

Reply:
and in all other get parameter functions that still exist

Reply:
I've left `radius` parameter to be uninitialized as want to developer to set it explicitly:
```
// Leave it not initialized: the will cause an error if it will not set
    nav2_util::declare_parameter_if_not_declared(
      node, polygon_name_ + ".radius", rclcpp::PARAMETER_DOUBLE);
```
This will throw an exception, if developer forgot to set a value for this parameter. We better to handle this, as error handling in catch will tell developer, in which polygon/source the parameter was not set.

In all other cases where try-catch blocks are remained, they are handling similar cases where ROS-parameters intentionally have no default values. Do you think, we need to change default values to something definite instead for all these cases?